src/services/TextRenderService.js

```diff
@@ -132,3 +138,66 @@ function renderLinkLayer(page, textLayer, viewport) {
     EventHandlerService.publish(PDFLEvents.onLinkLayerRendered, textLayer);
   });
 }
+
+/**
+ * Creates the page context where render the page.
+ *
+ * @param {canvas} canvas canvas of the pdf.js page
+ * @param {viewport} viewport target page viewport for the textlayer
+ * @returns {renderCtx} renderCtx, element where render the page
+ */
+export function getContext(canvas, viewport) {
+  const ctx = canvas.getContext("2d");
+  const renderCtx = {
+    canvasContext: ctx,
+    viewport: viewport,
+  };
+  return renderCtx;
+}
+
+/**
+ * Creates the page viewport and sets canvas size.
```

Comment:
It is a little weird to have a "get" function mutate objects.

Reply:
That's the way to have small functions working for both the render.